test/e2e/main_test.go

```diff
@@ -420,6 +416,20 @@ const (
 	prometheusOperatorServiceName = "prometheus-operator"
 )
 
+// TestFeatureGatedFeatures tests features that are behind feature gates.
+// Altough we'd like to have these tests green at all times, they are not
+// critical for the release process, therefore they can fail without blocking PRs.
```

Comment:
I might not agree 100% with this :)

For now, all feature gates are alpha so yes they could break anytime but in the future, some of them (hopefully) will become beta. Past experience also showed me that non-gating jobs are very likely to degrade over time.

Reply:
Fair enough, I did not think of beta graduation of features.

Should we split tests again in the future? (Beta are required to pass, Alpha not required)

Reply:
We can still merge without all required tests being green so I'd rather be strict from the start and avoid considering these tests as "less important".

Reply:
Fair point, changes adjusted